### PR TITLE
Check if an element should exist in the first place

### PR DESCRIPTION
**Pull request by zamattiac**
#### What matters to alternate check:

Either: One or the other exists
#### What matters to existential check:

Neither: If the loader isn't present, the final shouldn't be either
Both: If the loader is present, the final must be present

List of projects on an institution page:
- PASS: No 'fetching', no fg-file-links (they shouldn't exist)
- PASS: No 'fetching', fg-file-links
- FAIL: 'Fetching', no fg-file-links (they should exist)
- FAIL: 'Fetching', fg-file-links (many are loaded)

List of components on a project page:
- PASS: No containment, no render-node
- IMPOSSIBLE: No containment, render-node
- FAIL: Containment, no render-node
- PASS: Containment, render-node

**if loader is present** :

```
INSTITUTION: check if the final exists in the page
no fg-file-links (should exist): FAIL
some fg-file-links: PASS ANYWAY


PROJECT COMPONENTS: check if the final exists in the page
no nodes rendered: FAILURE
all nodes rendered: SUCCESS
```

_TLDR: REJECT PRESENT LOADER IF FINAL IS ABSENT_

**else** :

```
INSTITUTION: PASS
institution has no projects
projects are all loaded


COMP: PASS
no nodes to render
```

_TLDR: IGNORE THE ABSENCE OF THE LOADER_

### CODE REVIEW

File: verifier.py [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
-    def harvest_pages(self, json_filename, json_list):
+    def harvest_pages(self, json_dictionary, json_list):
         """
-        :param json_filename: The json file
+        :param json_dictionary: The dictionary created from the json file
         :param json_list: The list in the json file of found URLs
         :return: Null, but self.pages is populated.
         """
-        for url in json_list:
+        for url in json_list[:]:
```
Comment (zamattiac) [inline verifier.py:40]: This is the really important addition for some reason, which allows iteration to not skip when removing.

**Comment by sf2ne** [comment]
If we know the page doesn't have all its components loaded we shouldn't pass it anyway.

**Comment by sf2ne** [comment]
I'd say make this pretty specific. Overwrite the method for its specific sub classes. Verification can be a bit heavy handed, that's kind of its point. We need to be completely sure for each page.
